fitsio/src/headers/card.rs

```diff
@@ -0,0 +1,55 @@
+//! Fits HDU related code
+
+use fitsio_sys::{FLEN_COMMENT, FLEN_KEYWORD, FLEN_VALUE};
+
+use crate::errors::Result;
+use std::ffi::{c_char, CStr};
+
+/// Wraps a single header card
+#[derive(Debug)]
+pub struct Card {
+    pub(crate) name:       [i8;FLEN_KEYWORD as usize],
+    pub(crate) value:      [i8;FLEN_VALUE as usize],
+    pub(crate) comment:    [i8;FLEN_COMMENT as usize],
```

Comment:
I can see why these methods are here in this implementation (lazily converting the types if required, rather than eagerly converting the types at `Card` creation) however I prefer doing the work up front:
* the `Card` type is really a data object, with no inherent behaviour. As such, it really should just be a wrapper around the state
* if the user wants the `Card` then they likely want at least two of the fields (name and value) so most of the conversion work is not wasted
* the conversion is not exactly intensive so it probably not a performance issue right now.

Reply:
Name and comment will always be strings, but given that we don't know what type value is we cannot do the conversion upfront and the data structure must support all possible value types. This leaves us with two options: 

* Using a the `i8` array as-is; or
* Introducing a enum with variants of each type carrying a value.

I am still quite unfamiliar with the details of the `fitsio` API but I don't think we have that enum and the generic return types of `read_key` points to the first option with the buffer.

The struct would thus change to 
```rust
pub(crate) name:       String,
pub(crate) comment:    String,
pub(crate) value:      [i8;FLEN_VALUE as usize],
```
Thoughts?

Reply:
Or better yet:
```rust
pub(crate) name:       String,
pub(crate) comment:    Option<String>,
pub(crate) value:      [i8;FLEN_VALUE as usize],
```
... given that comments are optional.